### PR TITLE
add PostMessageTransport.js

### PR DESCRIPTION
**Pull request by AJDBenner**

### CODE REVIEW

File: src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                    window.addEventListener("message", function (event) {
+                        console.log("[Brackets LiveDev] Got message: " + event.data);
+                        if (self._callbacks && self._callbacks.message) {
+                            self._callbacks.message(event.data);
+                        }
+                    });
+                }
+            };
+    global._Brackets_LiveDev_Transport = postMessageTransport;
+}(this));
```
Comment (sedge) [inline src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js]: This file needs a newline

File: src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// This is a transport injected into the browser via a script that handles the low
+// level communication between the live development protocol handlers on both sides.
+// This transport provides a postMessage mechanism. It's injected separately from the
+// protocol handler so that the transport can be changed separately.
+
+(function (global) {
+    "use strict";
+    
+   var postMessageTransport = {
```
Comment (sedge) [inline src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js]: The indenting is all wonky here and below - make sure to compare it to the NodeSocketTransportRemote

File: src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                 * Sets the callbacks that should be called when various transport events occur. All callbacks
+                 * are optional, but you should at least implement "message" or nothing interesting will happen :)
+                 * @param {?{connect: ?function, message: ?function(string), close: ?function}} callbacks
+                 *      The callbacks to set.
+                 *      connect - called when a connection is established to Brackets
+                 *      message(msgStr) - called with a string message sent from Brackets
+                 *      close - called when Brackets closes the connection
+                 */
+                setCallbacks: function (callbacks) {
+                    if (!global._Brackets_LiveDev_Socket_Transport_URL) {
```
Comment (sedge) [inline src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js]: What's happening here? This looks like websocket code that survived the culling.

File: src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                    parent.window.postMessage(JSON.stringify({
+                        type: "message",
+                        message: msgStr
+                    }), "*");
+                },
+                
+                /** 
+                 * Establish postMessage connection.
+                 */
+                enable: function () {
```
Comment (sedge) [inline src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js]: I don't see any connection logic here. It looks like you just listen for message events and pass them off to the protocol (which I'm assuming is the one setting callbacks). Have you co-ordinated with Klever on this?

File: src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                        this._callbacks = callbacks;
+                    }
+                },
+                
+                /**
+                 * Sends a message over the transport.
+                 * @param {string} msgStr The message to send.
+                 */
+                send: function (msgStr) {
+                    parent.window.postMessage(JSON.stringify({
```
Comment (humphd) [inline src/LiveDevelopment/MultiBrowserImpl/transports/remote/PostMessageTransportRemote.js]: This can just be `parent.postMessage(...)`

**Comment by humphd** [comment]
Let's move this file into the extension we're doing vs. having it in Bracket's src.

**Pull request closed without merging**